test/cql-pytest/test_tools.py

```diff
@@ -259,3 +261,30 @@ def test_scylla_sstable_write(cql, test_keyspace, scylla_path, scylla_data_dir,
             actual_json = json.loads(actual_out)["sstables"]["anonymous"]
 
             assert actual_json == original_json
+
+
+@pytest.mark.parametrize("table_factory", [
+        simple_no_clustering_table,
+        simple_clustering_table,
+        clustering_table_with_collection,
+        clustering_table_with_udt,
+        table_with_counters,
+])
+def test_scylla_sstable_script(cql, test_keyspace, scylla_path, scylla_data_dir, table_factory):
+    scripts_path = os.path.realpath(os.path.join(__file__, '../../../tools/scylla-sstable-scripts'))
+    dump_script_path = os.path.join(scripts_path, 'dump.lua')
+    with scylla_sstable(table_factory, cql, test_keyspace, scylla_data_dir) as (schema_file, sstables):
+        dump_common_args = [scylla_path, "sstable", "dump-data", "--schema-file", schema_file, "--output-format", "json"]
+        script_common_args = [scylla_path, "sstable", "script", "--schema-file", schema_file, "--script-file", dump_script_path]
+
+        # without --merge
+        cxx_json = json.loads(subprocess.check_output(dump_common_args + sstables))
+        lua_json = json.loads(subprocess.check_output(script_common_args + sstables))
+
+        assert lua_json == cxx_json
```

Comment:
This is an excellent test to start with, but it might be tedious to maintain in the long run, if you ever want to improve one of the two implementations (the "dump-data" or the LUA script) you'll need to do the same for both. But we can cross this bridge when (or if) we reach it.

Reply:
One positive side of this test is I got to again thoroughly review my json output format and discovered some details that were in need of improvement.
Another is that we now have a stability test for the `dump-data` operation: if in the future we accidentally change the output format this test will flag it up.
Note that we don't plan to change the format too often, as it would possibly break any script people came up with to parse it.